### PR TITLE
Enabled rubocop

### LINKED ISSUE #813: Enable Rubocop
I think that since policial was dismantled, we have not enabled linting on this repo for a while. We need to get our code up to appropriate standards.

### PR DESCRIPTION
**Pull request by adfaris**
resolves #813 

add a rubocop gem to the project and run rubocop with the autocorrect option on and the before and after results are shown below. 

_Before:_
```127 files inspected, 382 offenses detected```

_After:_
```127 files inspected, 0 offenses detected```

Screen Shot 2020-04-10 at 11 03 48 AM

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
-  - https://shopify.github.io/ruby-style-guide/rubocop.yml
+inherit_gem:
+    rubocop-shopify: rubocop.yml
 
-LineLength:
+AllCops:
+  TargetRubyVersion: 2.4
   Exclude:
-    - test/**/*
+    - 'lib/generators/**/*'
```
Comment (vfonic) [inline .rubocop.yml:7]: Hey @adfaris!

Sorry for coming late to the game.

Why are we excluding `'lib/generators/**/*'` here?

Comment (adfaris) [reply .rubocop.yml:7]: Rubocop doesn't play with the template files with erb in them. https://github.com/Shopify/shopify_app/blob/f95aac7fc02412e0f6d57bcdf425cb2df1414644/lib/generators/shopify_app/add_webhook/templates/webhook_job.rb#L1

Perhaps we can be more specific here.

Comment (vfonic) [reply .rubocop.yml:7]: Ah I see!

Yeah, maybe `'lib/generators/**/templates/*'` or similar? I haven't tried this out.

Comment (vfonic) [reply .rubocop.yml:7]: Actually I think the real issue here is that those are not `.rb` files.

Those are `.rb.tt` files, I believe. Take a look at how rails calls them:
https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/controller/templates/controller.rb.tt

/cc @tanema

Comment (adfaris) [reply .rubocop.yml:7]: I made these changes in this PR: 

https://github.com/Shopify/shopify_app/pull/959

**Pull request merged**